analyzers/tests/SonarAnalyzer.UnitTest/TestCases/EventHandlerDelegateShouldHaveProperArguments.CSharp11.cs

```diff
@@ -0,0 +1,31 @@
+﻿using System;
+
+namespace Tests.Diagnostics
+{
+    public interface IMyInterface
+    {
+        public static event EventHandler SomeStaticEvent;
```

Comment:
Please add test cases with virtual and abstract events.

Reply:
Actually with abstract events I think it is not possible to make a test case however with virtual it is. Here you go:
```
public interface IMyInterface
    {
        public static virtual event EventHandler VirtualEvent;
        public static void VirtualMethod<T>(object sender, EventArgs e) where T : IMyInterface
        {
            T.VirtualEvent.Invoke(null, null);            // Noncompliant
            VirtualEvent.Invoke(null, null);            // Noncompliant
            T.VirtualEvent.Invoke(null, e);            // Compliant
            VirtualEvent.Invoke(null, e);            // Compliant
        }
    }
```